File: downloadSorterCLI/AttributeExtractor.m
//
//  WhereFromExtractor.m
//  DownloadSorterSwift
//
//  Created by Wolfgang Lutz on 24.04.15.
//  Copyright (c) 2015 Wolfgang Lutz. All rights reserved.
//

#import "AttributeExtractor.h"

@implementation AttributeExtractor

+ (NSArray*) getWhereFromForPath:(NSString*) path{
    MDItemRef item = MDItemCreate( kCFAllocatorDefault, (CFStringRef)CFBridgingRetain(path) );
    
    CFArrayRef list = MDItemCopyAttributeNames( item );
    
    NSDictionary *resDict = (NSDictionary *)CFBridgingRelease(MDItemCopyAttributes( item, list ));
    CFRelease( list );
    CFRelease( item );
    
    return [resDict objectForKey:@"kMDItemWhereFroms"];
};

@end

File: downloadSorterCLI/Utils/AttributeExtractor.swift
//
//  AttributeExtractor.swift
//  downloadSorter
//
//  Created by Wolfgang Lutz on 02.08.17.
//  Copyright © 2017 Number 42. All rights reserved.
//

import Foundation

struct AttributeExtractor {
  static func getWhereFromsFromFile(at path: String) -> [String] {
    let item: MDItem? = MDItemCreate(kCFAllocatorDefault, path as CFString)
    let list: CFArray = MDItemCopyAttributeNames(item)
    let resDict: [AnyHashable: Any]? = (MDItemCopyAttributes(item, list) as? [AnyHashable: Any])
    return resDict?["kMDItemWhereFroms"] as? [String] ?? [String]()
  }
}
